cmd/application.go

```diff
@@ -139,6 +141,55 @@ var applicationSetCmd = &cobra.Command{
 	},
 }
 
+var applicationDescribeCmd = &cobra.Command{
+	Use:   "describe",
+	Short: "describe the given application",
+	Example: `  # Describe someapp application,
+  odo app describe someapp
+	`,
+	Run: func(cmd *cobra.Command, args []string) {
+		client := getOcClient()
+		var currentApplication string
+		if len(args) == 0 {
+			var err error
+			currentApplication, err = application.GetCurrent(client)
+			checkError(err, "")
+		} else {
+			currentApplication = args[0]
+			//Check whether application exist or not
+			exists, err := application.Exists(client, currentApplication)
+			checkError(err, "")
+			if !exists {
+				fmt.Printf("application with the name %s does not exist\n", currentApplication)
+				os.Exit(1)
+			}
+		}
+		fmt.Printf("Application %s has:", currentApplication)
+		//Project
+		currentProject := project.GetCurrent(client)
+		// List of Component
+		componentList, err := component.List(client, currentApplication, currentProject)
```

Comment:
If `len(componentList) == 0`, print a message saying that "This application has no components deployed."